apps/server/src/setup/loadDb.ts

```diff
@@ -44,35 +47,49 @@ const populateDb = (directory: string, filename: string): string => {
 };
 
 /**
- * @description parses a json file to the adapter
- * It will create an empty file from the model if the parsing fails
+ * Handles a corrupted fle by copying it to a corrupted folder
+ * Eventual recovered data will be added to a new file
  */
-const parseDatabase = async (fileToRead: string, adapterToUse: Low<DatabaseModel>) => {
+async function handleCorruptedDb(filePath: string) {
   try {
-    // this will throw if file is not valid
-    parseProjectFile(fileToRead);
-    await adapterToUse.read();
-  } catch (error) {
-    adapterToUse.data = dbModel;
+    const fileName = getFileNameFromPath(filePath);
+    const newFilePath = join(resolveCorruptedFilesDirectory, fileName);
+
+    ensureDirectory(resolveCorruptedFilesDirectory);
+    copyFileSync(filePath, newFilePath);
+    await renameProjectFile(fileName, nameRecovered(fileName));
+  } catch (_) {
+    /* we do not handle errors here */
   }
-
-  return parseJson(adapterToUse.data);
-};
+}
 
 /**
  * @description loads ontime db
  */
 async function loadDb(directory: string, filename: string) {
   const dbInDisk = populateDb(directory, filename);
 
-  const adapter = new JSONFile<DatabaseModel>(dbInDisk);
-  const db = new Low(adapter, dbModel);
+  let newData: DatabaseModel = dbModel;
+  let errors = [];
+
+  try {
+    const maybeProjectFile = parseProjectFile(dbInDisk);
+    const result = parseJson(maybeProjectFile);
+    newData = result.data;
+    errors = result.errors;
+  } catch (error) {
+    consoleError(`Unable to parse project file: ${getErrorMessage(error)}`);
+    handleCorruptedDb(dbInDisk);
+  } finally {
+    if (errors.length > 0) {
+      handleCorruptedDb(dbInDisk);
+    }
```

Comment:
I think somewhere here we should the load the recovered file?
Right now it looks like we stay on the broken one

https://github.com/cpvalente/ontime/assets/19875125/c784a92a-57d0-4e43-a85a-a84c047ed600